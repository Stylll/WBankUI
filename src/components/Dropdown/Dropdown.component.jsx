import React from 'react';
import PropTypes from 'prop-types';

import './Dropdown.styles.scss';

const Dropdown = ({
  name, value, placeholder, onChange, info,
  error, required, styles, options, disabled,
  optionsValueProperty, optionsTitleProperty,
  dataTestId,
}) => {
  const iconClassName = disabled ? 'chevy disabled' : 'chevy';

  return (
    <div className="drop-down">
      <select className="select" required={required}
        onChange={onChange} name={name} value={value} style={styles}
        disabled={disabled} data-testid={dataTestId}
      >
        <option value="">{placeholder}</option>
        {options.map(
          (option, index) => (
            <option
              value={option[optionsValueProperty]}
              key={index}
            >
              {option[optionsTitleProperty]}
            </option>),
        )}
      </select>
      <span className={iconClassName}>
        &or;
      </span>
      {!error && <span className="info">{info}</span>}
      {error && <span className="error" id={`${name}-error`}>{error}</span>}
    </div>
  );
};

Dropdown.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  styles: PropTypes.object,
  disabled: PropTypes.bool,
  options: PropTypes.array.isRequired,
  optionsTitleProperty: PropTypes.string.isRequired,
  optionsValueProperty: PropTypes.string.isRequired,
  dataTestId: PropTypes.string,
};

Dropdown.defaultProps = {
  disabled: false,
  dataTestId: 'dropdown',
};

export default Dropdown;
