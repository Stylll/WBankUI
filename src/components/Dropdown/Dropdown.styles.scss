@import '../../partials/colors';

.drop-down {
    position: relative;
    width: 100%;
    select {
        width: 100%;
        height: 30px;
        border: 0px;
        border-bottom: 1px solid $gray;
        border-radius: 0px;
        outline: none;
        font-size: 14px;
        font-weight: 600;
        color: $primary-color;
        padding: 2px 4px;
        background-color: $white;
        -moz-appearance: none;
        -webkit-appearance: none;
        cursor: pointer;
        &:focus {
            border-color: $primary-color;
        }
        &:disabled {
            cursor: default;
            color: $gray;
        }
    }
    span.chevy {
        position: absolute;
        right: 5px;
        top: 10px;
        font-size: 12px;
        color: $primary-color;
        &.disabled {
            color: $gray;
        }
    }
    .info {
        font-size: 12px;
        color: $gray;
    }
    .error {
        font-size: 12px;
        color: $red;
    }
}
