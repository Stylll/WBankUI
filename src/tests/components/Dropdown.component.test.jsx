import { render, fireEvent } from '@testing-library/react';
import React from 'react';

import Dropdown from '../../components/Dropdown/Dropdown.component';

describe('Dropdown Component', () => {
  const currencies = [
    { id: 'cad', name: 'CAD' },
    { id: 'usd', name: 'USD' },
    { id: 'mxn', name: 'MXN' },
  ];
  it('should render properly', () => {
    const { baseElement } = render(<Dropdown
            name="currency"
            value="usd"
            error=""
            onChange={() => {}}
            options={currencies}
            optionsTitleProperty="name"
            optionsValueProperty="id"
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should call onChange method when dropdownlist is changed', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(<Dropdown
        name="currency"
        value="usd"
        error=""
        onChange={onChange}
        options={currencies}
        optionsTitleProperty="name"
        optionsValueProperty="id"
        dataTestId="dropdownlist-test"
    />);

    fireEvent.change(getByTestId('dropdownlist-test'), { target: { value: 'cad' } });
    expect(onChange).toHaveBeenCalled();
  });
});
