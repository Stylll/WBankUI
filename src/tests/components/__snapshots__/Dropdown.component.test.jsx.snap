// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component should render properly 1`] = `
<body>
  <div>
    <div
      class="drop-down"
    >
      <select
        class="select"
        data-testid="dropdown"
        name="currency"
      >
        <option
          value=""
        />
        <option
          value="cad"
        >
          CAD
        </option>
        <option
          value="usd"
        >
          USD
        </option>
        <option
          value="mxn"
        >
          MXN
        </option>
      </select>
      <span
        class="chevy"
      >
        âˆ¨
      </span>
      <span
        class="info"
      />
      
    </div>
  </div>
</body>
`;
